doc/rst/technotes/extern.rst

```diff
@@ -167,6 +157,11 @@ pointed-to type T should be one that is supported in C interoperability if the
 c_ptr(T) is used for C interoperability. The c_ptr(T) type supports
 indexing to get a reference to the i'th element (starting from 0).
 
+The c_ptr(void) type is an opaque C pointer, corresponding to ``void*`` in C.
+Since the type is void (unknown), there is no way to dereference this pointer.
+Normally, a C function will return the void pointer, which will be passed to
```

Comment:
I would write:

```
The ``c_ptr(void)`` type represents the C ``void*`` type. There is no way to dereference such a pointer in Chapel code without first casting the pointer one with a different underlying type.
```

I'm not sure I would have the sentence about it not being possible to construct such a pointer. Can I just write:

```chapel
var x: c_ptr(void) = c_ptrTo(foo);
```
If so then I'd say that `c_ptr(void)` just obeys the same restrictions as other pointers with the dereferencing caveat.

I also think I'd drop the sentence about typical usage.

Reply:
Yes, that Chapel code creating a `c_ptr(void)` would work, via implicit conversion from whatever `c_ptrTo(foo)` returned. I did remove the sentence about not being able to construct one.

I agree with removing the typical usage sentence as well.